server/src/main/java/org/elasticsearch/index/codec/tsdb/DocValuesForUtil.java

```diff
@@ -25,32 +29,70 @@ public DocValuesForUtil(int blockSize) {
         this.blockSize = blockSize;
     }
 
-    void encode(long[] in, int bitsPerValue, DataOutput out) throws IOException {
+    public static int roundBits(int bitsPerValue) {
+        if (bitsPerValue > 32 && bitsPerValue <= BITS_IN_FIVE_BYTES) {
+            return BITS_IN_FIVE_BYTES;
+        } else if (bitsPerValue > BITS_IN_FIVE_BYTES && bitsPerValue <= BITS_IN_SIX_BYTES) {
+            return BITS_IN_SIX_BYTES;
+        } else if (bitsPerValue > BITS_IN_SIX_BYTES && bitsPerValue <= BITS_IN_SEVEN_BYTES) {
+            return BITS_IN_SEVEN_BYTES;
+        }
+        return bitsPerValue;
+    }
+
+    void encode(long[] in, int bitsPerValue, final DataOutput out) throws IOException {
         if (bitsPerValue <= 24) { // these bpvs are handled efficiently by ForUtil
             forUtil.encode(in, bitsPerValue, out);
         } else if (bitsPerValue <= 32) {
```

Comment:
should we round numbers of bits between 25 and 32 to 32 and have an equality check here?

Reply:
I think this is not strictly necessary. I think it is more a matter of being consistent between what we write as the `bitsPerValue` and what we do when encoding and decoding. I mean if we write `bitsPerValue = 27` for instance...we encode and decode using 32 bits even if we wrote 27 as `bitsPerValue`. If we do what you suggest we write the actual value 32 and do the encode and decode using 21 bits.